test/framework/http.go

```diff
@@ -72,7 +74,30 @@ func WaitForSuccessfulHTTPRequest(namespace string, requestInfo HTTPRequestInfo,
 
 // ExecuteHTTPRequest executes an HTTP request
 func ExecuteHTTPRequest(namespace string, requestInfo HTTPRequestInfo) (*http.Response, error) {
-	return ExecuteHTTPRequestC(&http.Client{}, namespace, requestInfo)
+	// Setup a retry in case the first time it did not work
+	retry := 0
+	var resp *http.Response
+	var err error
+	for retry < config.GetHTTPRetryNumber() {
+		resp, err := ExecuteHTTPRequestC(&http.Client{}, namespace, requestInfo)
+		if err == nil && checkHTTPResponseSuccessful(namespace, resp) {
+			return resp, err
+		} else if err != nil {
+			GetLogger(namespace).Warnf("Error while making http call: %v", err)
+		} else if resp != nil {
+			if resp.StatusCode == 500 {
+				// In case of 500, it is server error and we don't need to call again
+				return resp, err
+			}
+			GetLogger(namespace).Warnf("Http call was not successful. Response code: %d", resp.StatusCode)
```

Comment:
Maybe we could first log the message and then in case it is 500, return. Now it won't log the code if it is 500.

Reply:
updated